packages/app-mobile/components/NoteList.tsx

```diff
@@ -91,7 +92,13 @@ class NoteListComponent extends Component<NoteListProps> {
 				keyExtractor={item => item.id}
 			/>;
 		} else {
-			if (!Folder.atLeastOneRealFolderExists(this.props.folders)) {
+			const noteSourceParentId = this.props.notesSource ? JSON.parse(this.props.notesSource).parentId : null;
+			const noteSourceFolder = this.props.folders.find(folder => folder.id === noteSourceParentId);
+
+			if (noteSourceParentId === getTrashFolderId() || (noteSourceFolder?.deleted_time !== 0)) {
```

Comment:
Please use the utility function `itemIsInTrash()`

Reply:
done

Reply:
There seems to be some issue in the implementation of `itemIsInTrash()`. From what I understood from its code, it was intended to return `true` for trash folder and its subfolders, but currently it's only returning `true` for the sub folders of the trash folder and `false` for the trash folder itself. 

The function returns at `if (!isTrashableNoteOrFolder(item)) return false;` when the item is the trash folder as `isTrashableNoteOrFolder` returns `false` for the trash folder. Thus it never reaches the end `return item.id === getTrashFolderId() || !!item.deleted_time;` where it was supposed to return `true` for the trash folder

Reply:
The `isTrashableNoteOrFolder` check exists to prevent the `checkObjectHasProperties` assertion to fail unnecessarily (particularly in the case where `deleted_time` may be `undefined`).

One possible fix could be to move the `item.id === getTrashFolderId()` before the `isTrashableNoteOrFolder` check. For example,

<details><summary>diff</summary>

```diff
diff --git a/packages/lib/services/trash/index.ts b/packages/lib/services/trash/index.ts
index 35fc2c890..a9b1590bb 100644
--- a/packages/lib/services/trash/index.ts
+++ b/packages/lib/services/trash/index.ts
@@ -84,11 +84,13 @@ export const getTrashFolderIcon = (type: FolderIconType): FolderIcon => {
 
 export const itemIsInTrash = (item: FolderEntity | NoteEntity) => {
 	if (!item) return false;
-	if (!isTrashableNoteOrFolder(item)) return false;
 
-	checkObjectHasProperties(item, ['id', 'deleted_time']);
+	checkObjectHasProperties(item, ['id']);
+	if (item.id === getTrashFolderId()) return true;
+	if (!isTrashableNoteOrFolder(item)) return false;
 
-	return item.id === getTrashFolderId() || !!item.deleted_time;
+	checkObjectHasProperties(item, ['deleted_time']);
+	return !!item.deleted_time;
 };
 
 export const getRestoreFolder = async () => {
```

</details>

An alternate fix could be to add `deleted_time` to the definition of the conflict folder and remove the check entirely. For example,

<details><summary>diff</summary>

```diff
diff --git a/packages/lib/models/Folder.ts b/packages/lib/models/Folder.ts
index 38d163b6d..4602a9e77 100644
--- a/packages/lib/models/Folder.ts
+++ b/packages/lib/models/Folder.ts
@@ -181,6 +181,7 @@ export default class Folder extends BaseItem {
 			user_updated_time: now,
 			share_id: '',
 			is_shared: 0,
+			deleted_time: 0,
 		};
 	}
 
diff --git a/packages/lib/services/trash/index.ts b/packages/lib/services/trash/index.ts
index 35fc2c890..225df6c2a 100644
--- a/packages/lib/services/trash/index.ts
+++ b/packages/lib/services/trash/index.ts
@@ -84,10 +84,7 @@ export const getTrashFolderIcon = (type: FolderIconType): FolderIcon => {
 
 export const itemIsInTrash = (item: FolderEntity | NoteEntity) => {
 	if (!item) return false;
-	if (!isTrashableNoteOrFolder(item)) return false;
-
 	checkObjectHasProperties(item, ['id', 'deleted_time']);
-
 	return item.id === getTrashFolderId() || !!item.deleted_time;
 };
 

```

</details>